contracts/peg/BasePSM.sol

```diff
@@ -0,0 +1,339 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./../pcv/PCVDeposit.sol";
+import "./IBasePSM.sol";
+import "./../refs/OracleRef.sol";
+import "../Constants.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+
+abstract contract BasePSM is IBasePSM, OracleRef, PCVDeposit {
+    using Decimal for Decimal.D256;
+    using SafeCast for *;
+    using SafeERC20 for IERC20;
+
+    /// @notice the amount of reserves to be held for redemptions
+    uint256 public override reservesThreshold;
+
+    /// @notice the PCV deposit target
+    IPCVDeposit public override surplusTarget;
+
+    /// @notice the token this PSM will exchange for VOLT
+    IERC20 public immutable override underlyingToken;
+
+    /// @notice struct for passing constructor parameters related to OracleRef
+    struct OracleParams {
+        address coreAddress;
+        address oracleAddress;
+        address backupOracle;
+        int256 decimalsNormalizer;
+        bool doInvert;
+    }
+
+    /// @notice constructor
+    /// @param params PSM constructor parameter struct
+    constructor(
+        OracleParams memory params,
+        uint256 _reservesThreshold,
+        IERC20 _underlyingToken,
+        IPCVDeposit _surplusTarget
+    )
+        OracleRef(
+            params.coreAddress,
+            params.oracleAddress,
+            params.backupOracle,
+            params.decimalsNormalizer,
+            params.doInvert
+        )
+    {
+        underlyingToken = _underlyingToken;
+
+        _setReservesThreshold(_reservesThreshold);
+        _setSurplusTarget(_surplusTarget);
+        _setContractAdminRole(keccak256("PSM_ADMIN_ROLE"));
```

Comment:
Do you have a `TribeRoles.sol` or similar that you can pull this role signature from? Rather than manually doing `keccak256("PSM_ADMIN_ROLE")`